agent/internal/metrics.go

```diff
@@ -161,7 +161,27 @@ func (a *Agent) addUserEvent(event userEventFace) {
 		}
 	case *signupUserEvent:
 		store = a.metricsMng.get("sdk-signup")
+	default:
+		return
 	}
 
 	store.add(event)
 }
+
+type WhitelistedIP struct {
+	MatchedWhitelistEntry string
+}
+
+func (m WhitelistedIP) bucketID() (string, error) {
+	return m.MatchedWhitelistEntry, nil
+}
+
+func (a *Agent) addWhitelistEvent(matchedWhitelistEntry string) {
+	if a.config.Disable() || a.metricsMng == nil {
+		// Agent is disabled or not yet initialized
+		return
+	}
+	a.metricsMng.get("whitelisted").add(WhitelistedIP{
```

Comment:
the metric is enabled/disabled through a feature named `whitelisted_metric`

Reply:
Adding this to this to the backlog for now.